### PR TITLE
Adds JEI integration for animal herders (and the beekeeper)

### PR DESCRIPTION
**Pull request by uecasm**
# Changes proposed in this pull request:
- Adds JEI integration for animal herders (and the beekeeper), including loot drops
image
- Adds a module that could theoretically be used for common herder AI lookup tasks (to keep things in one place, and have them queryable externally too), but does not actually do that just yet.

Review please

### CODE REVIEW

**Review by Raycoms — APPROVED** [review]

**Comment by uecasm** [comment]
FYI I did get a little whimsical with some of the descriptions in `en_us.json` so you might want to give those extra review. 😉

File: src/main/resources/assets/minecolonies/lang/en_us.json [hunk 1]
```diff
@@ -1553,8 +1553,11 @@
   "com.minecolonies.coremod.jei.chancenegskill.tip": "§e%d%% base chance §c- skill modifier",
   "com.minecolonies.coremod.jei.conditions.tip": "§bSpecial conditions apply",
   "com.minecolonies.coremod.jei.baker": "Bakes bread, and other things at higher levels. Can be taught extra recipes starting at level 3.",
+  "com.minecolonies.coremod.jei.beekeeper": "Takes care of beehives, harvesting either honey or honeycomb.",
```
Comment (Raycoms) [inline src/main/resources/assets/minecolonies/lang/en_us.json:1556]: @ravenbuilder934

Comment (uecasm) [reply src/main/resources/assets/minecolonies/lang/en_us.json:1556]: I came _very_ close to making a "not the bees!" reference there, but I managed to refrain.  I may have been less successful in other places.

**Review by marchermans — APPROVED** [review]
Good job!

**Review by someaddons — APPROVED** [review]

**Pull request merged**